pep-0517.txt

```diff
@@ -786,23 +787,25 @@ any change can go live, and that any changes will necessarily be
 restricted to new releases.
 
 One specific consequence of this is that in this PEP, we're able to
-make the ``prepare_wheel_metadata`` command optional. In our design, this
-can easily be worked around by a tool like ``pip``, which can put code
-in its subprocess runner like::
-
-    def prepare_wheel_metadata(output_dir, config_settings):
-         if hasattr(backend, "prepare_wheel_metadata"):
-             backend.prepare_wheel_metadata(output_dir, config_settings)
-         else:
-             backend.build_wheel(output_dir, config_settings)
-             touch(output_dir / "PIP_ALREADY_BUILT_WHEELS")
-             unzip_metadata(output_dir/*.whl)
+make the ``prepare_metadata_for_build_wheel`` command optional. In our design,
+this can be readily handled by build frontends, which can put code in
+their subprocess runner like::
+
+    def prepare_metadata_for_build_wheel(output_dir, config_settings):
+        if hasattr(backend, "prepare_metadata_for_build_wheel"):
+            backend.prepare_metadata_for_build_wheel(output_dir, config_settings)
+        else:
+            wheel_fname = backend.build_wheel(output_dir, config_settings)
+            (output_dir / "ALREADY_BUILT_WHEEL").write_text(wheel_fname)
+            unzip_metadata(output_dir / wheel_fname)
 
     def build_wheel(output_dir, config_settings, metadata_dir):
-         if os.path.exists(metadata_dir.parent / "PIP_ALREADY_BUILT_WHEELS"):
-             copy(metadata_dir.parent / *.whl, output_dir)
-         else:
-             backend.build_wheel(output_dir, config_settings, metadata_dir)
+        already_built = metadata_dir.parent / "ALREADY_BUILT_WHEEL"
```

Comment:
This implies that we're passing pathlib objects to hooks. I personally like working with pathlib, but I think that specifying that would get messy for Python versions which don't have the pathlike protocol, and even worse for those which don't have pathlib in the standard library.

So I'm inclined to use plain string paths throughout the spec and all examples.